File: WEEKLYCODE#3/sansa_and_xor/C/sansa_and_xor.c
#include <stdio.h>

int main() {
    int t,i;
    scanf("%d",&t);
    //for each test case
    while(t--){
        int n,res=0;
        //taking input
        scanf("%d",&n);
        int ar[n];
        for(i=0;i<n;i++)
            scanf("%d",&ar[i]);
        //if n is odd ,then result will be xor of even indexed elements otherwise 0
        if(n%2!=0)
            for(i=0;i<n;i+=2)
                res=res^ar[i];
       //printing result
       printf("%d\n",res);
    }
    return 0;
}

File: WEEKLYCODE#3/sansa_and_xor/Java/sansa_and_xor.java
import java.io.*;
import java.util.*;
import java.text.*;
import java.math.*;
import java.util.regex.*;

public class Solution
{
    public static void main(String[] args) {
        Scanner input = new Scanner(System.in);
        int t = input.nextInt();
        //for each test case
            for(int j=0;j<t;j++) {
                int n = input.nextInt();
                int res = 0,i;
                int a[] = new int[n];
                //taking input
                for ( i = 0; i < n; i++) 
                    a[i] = input.nextInt();
                //if n is odd ,then result will be xor of even indexed elements otherwise 0
                if(n%2 != 0)
                    for( i = 0; i < n; i += 2)
                        res ^= a[i];
                //printing result
                System.out.println(res);
            }
    }
}
